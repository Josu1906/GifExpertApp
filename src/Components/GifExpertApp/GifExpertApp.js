import React, {useState} from "react";
import AddCategory from "../AddCategory/AddCategory";
import {GifGrid} from "../GifGrid/GifGrid"
//rafc para poner todo simplificado

const GifExpertApp = () => {
    // const categories = ['Elon Musk', 'Thanos', 'Linterna Verde', 'Spider-man', 'Batman', 'Ironman'];
    // const [categories, setCategories] = useState(['Elon Musk', 'Thanos', 'Linterna Verde', 'Spider-man', 'Batman', 'Ironman']);
    const [categories, setCategories] = useState([]);
    // const handelBand = () => {
    //     return setCategories([...categories, 'Josue']);
    // }
    return (
    <>
    <h2>GifExpertApp</h2>
    <hr></hr>
    <AddCategory setCategories={setCategories}/>
    {/* <button onClick={handelBand}>Agregar</button> */}
    <ul>
        {
            categories.map(categorie => {
                // return <li key={categorie}>{categorie}</li> //key le sirve a React para saber que item del array a sido eliminado o cambiado
                return <GifGrid key={categorie} category={categorie}/>
            })
        }
    </ul>
    </>
)}

export default GifExpertApp;
