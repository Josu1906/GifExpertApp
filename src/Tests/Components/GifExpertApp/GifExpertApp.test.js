import { shallow } from 'enzyme';
import GifExpertApp from '../../../Components/GifExpertApp/GifExpertApp';
    
describe('Pruebas en GifExpertApp', () => {
    test('Mostrar el contenido', () => {
        const wrapper = shallow(<GifExpertApp />)
        expect(wrapper).toMatchSnapshot();
    })
})