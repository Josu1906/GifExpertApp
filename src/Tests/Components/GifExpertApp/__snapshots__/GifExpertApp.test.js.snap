// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifExpertApp Mostrar el contenido 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <hr />
  <AddCategory
    setCategories={[Function]}
  />
  <ul />
</Fragment>
`;
